ros/wavemap_ros/src/input_handler/pointcloud_undistorter.cc

```diff
@@ -0,0 +1,103 @@
+#include "wavemap_ros/input_handler/pointcloud_undistorter.h"
+
+#include <wavemap_ros_conversions/time_conversions.h>
+
+namespace wavemap {
+PointcloudUndistorter::Result PointcloudUndistorter::undistortPointcloud(
+    GenericStampedPointcloud& stamped_pointcloud,
+    PosedPointcloud<>& undistorted_pointcloud, const std::string& fixed_frame) {
+  // Get the time interval
+  const uint64_t start_time = stamped_pointcloud.getStartTime();
+  const uint64_t end_time = stamped_pointcloud.getEndTime();
+
+  // Calculate the step size for the undistortion transform buffer
+  const auto& points = stamped_pointcloud.getPoints();
+  constexpr int kNumTimeIntervals = 400;
```

Comment:
Why 400 here? Did this just happen to be good enough for most cases and therefore there's no reason to reconfigure it?

Reply:
Thanks for pointing this out. Thinking about it more, this is way higher than necessary (although it didn't seem to cause noticeable overhead). I'll update it to a more reasonable value and make it configurable.